nifi-api/src/main/java/org/apache/nifi/expression/ExpressionLanguageScope.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.expression;
+
+/**
+ * Indicates the scope of expression language on a property descriptor
```

Comment:
would be good to add comments here and/or with each scope but they're effectively hierarchical.

NONE -> VARIABLE_REGISTRY -> FLOWFILE

If something supports flowfile it supports both flowfile attributes and variable registry.  If something supports variable registry then it does not consider flowfile attributes.  Further, when supporting the variable registry there is an implied scope of 'variables defined in the closest process group and then up to the higher group and so on until the root group and finally the nifi.properties variables are evaluated and then environment variables/etc..'

Reply:
Right, I added some comments to reflect how scope are hierarchically used.